### ISSUE TITLE
chi2 for points marked as bad

### ISSUE THREAD

**Issue by rpoleski** [rpoleski/MulensModel#41]
Sample code:

```python
import numpy as np

import MulensModel as mm


data = mm.MulensData(file_name="data/photometry_files/OB08092/phot_ob08092_O4.dat")
bad = np.zeros(data.n_epochs, dtype=bool)
bad[1] = True
data.bad = bad

model = mm.Model({"t_0": 5379.571, "u_0": 0.523, "t_E": 17.94})
event = mm.Event(data, model)
print(event.get_chi2_per_point()[0][:3])
print(event.fits[0]._data_magnification[:3])
```

Returns:

```python
[2.65833274e+00 2.46814081e+04 9.53673023e-03]
[1.00107478 0.         1.00136059]
```
and both shows the problem (chi2 of 24.681,4 for a point marked as bad) and it's cause (assumption of magnification = 0). I propose to add a new parameter to `Event.get_chi2_per_point()` that will control if the returned values for bad data are some selected value (`0`, `None`, or `np.inf`) or are calculated as all the other ones. What are opinions on that?

**Comment by rpoleski** [comment]
@jenniferyee can you comment on that?

**Comment by jenniferyee** [comment]
The problem is that we don't always calculate the magnification for points
marked 'bad'. Several functions have a keyword 'bad', where bad=True means
"include bad points in the calculation." We can add this to
get_chi2_per_point():

event.get_chi2_per_point(bad=True)

But also part of the underlying issue is that magnification --> 0 for bad
points (unless get_magnification(bad=True)). Maybe the default value should
be np.nan instead. That should produce more transparent behavior.


--------------------------------------------------------------
Dr. Jennifer C. Yee
(she/her)

Office: P-341
Center for Astrophysics | Harvard & Smithsonian
60 Garden St, MS-15
Cambridge, MA 02138
--------------------------------------------------------------

**Comment by rpoleski** [comment]
I know how to solve these issues and tested these solutions:
1) Adding `bad` option to `Event.get_chi2_per_point()`: simple change inside that function
`self.fit_fluxes()` -> `self.fit_fluxes(bad=bad)`
works.
2) Setting the chi2 values for bad points (if bad=False): a change of the loop
```python
for (i, dataset) in enumerate(self.datasets):
    chi2_per_point.append(self.fits[i].chi2_per_point)
```
===>>>
```python
for (i, dataset) in enumerate(self.datasets):
    chi2 = self.fits[i].chi2_per_point
    chi2[dataset.bad] = XXX
    chi2_per_point.append(chi2)
```
were `XXX` is `0`, `np.nan` etc. (this could even be a parameter of this function)

Of course, applying both changes requires additional if statements. 

I propose to use `0` for XXX, so that sum of chi2 per point values is the same as chi2. @jenniferyee do you agree on that? And on applying both changes?

**Comment by jenniferyee** [comment]
I don't agree with using 0 for XXX, because it is possible for 0 to be the chi2 for a real data point.

For the sum of chi2 per point, is this a problem because the user might do this? or because we do this somewhere in the code?  If we use np.nan, we can do the sum using np.nansum(), but I suppose the user might not know to do this.

**Comment by rpoleski** [comment]
Solved in a73c940 (version 2.10).
